src/main/java/com/impactupgrade/nucleus/model/CrmImportEvent.java

```diff
@@ -558,35 +558,24 @@ public static BigDecimal getAmount(Map<String, String> data, String columnName)
   }
 
   // TODO: Hate this code -- is there a lib that can handle it in a forgiving way?
-  private static Calendar getDate(Map<String, String> data, String columnName) {
-    String key = null;
-
+  private static ZonedDateTime getDate(Map<String, String> data, String columnName, Environment env) {
     if (data.containsKey(columnName + " dd/mm/yyyy")) {
-      key = columnName + " dd/mm/yyyy";
+      return getZonedDateFromDateString(data.get(columnName + " dd/mm/yyyy"), env.getConfig().timezoneId, "d/M/yyyy");
     } else if (data.containsKey(columnName + " dd-mm-yyyy")) {
-      key = columnName + " dd-mm-yyyy";
+      return getZonedDateFromDateString(data.get(columnName + " dd-mm-yyyy"), env.getConfig().timezoneId, "d-M-yyyy");
     } else if (data.containsKey(columnName + " mm/dd/yyyy")) {
-      key = columnName + " mm/dd/yyyy";
+      return getZonedDateFromDateString(data.get(columnName + " mm/dd/yyyy"), env.getConfig().timezoneId, "M/d/yyyy");
     } else if (data.containsKey(columnName + " mm/dd/yy")) {
-      key = columnName + " mm/dd/yy";
+      return getZonedDateFromDateString(data.get(columnName + " mm/dd/yy"), env.getConfig().timezoneId, "M/d/yy");
     } else if (data.containsKey(columnName + " mm-dd-yyyy")) {
-      key = columnName + " mm-dd-yyyy";
+      return getZonedDateFromDateString(data.get(columnName + " mm-dd-yyyy"), env.getConfig().timezoneId, "M-d-yyyy");
     } else if (data.containsKey(columnName + " yyyy-mm-dd")) {
-      key = columnName + " yyyy-mm-dd";
-    }
-
-    if (Strings.isNullOrEmpty(key) || Strings.isNullOrEmpty(data.get(key))) {
-      return null;
+      return getZonedDateFromDateString(data.get(columnName + " yyyy-mm-dd"), env.getConfig().timezoneId, "yyyy-M-d");
     }
 
-    try {
-      Calendar c = Calendar.getInstance();
-      c.setTime(new SimpleDateFormat("dd/MM/yyyy").parse(data.get(key)));
```

Comment:
@VSydor This line was the original, root issue that kicked all this off. We were ignoring the format selected above. But to fix this, I went further and got rid of Calendar in favor of ZonedDateTime + using the org's timezone.

Reply:
Switch to ZonedDateTime is a very nice move!